src/sql/parts/notebook/models/notebookModel.ts

```diff
@@ -451,7 +451,10 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			let newConnectionProfile = new ConnectionProfile(this.notebookOptions.capabilitiesService, newConnection);
 			this._activeConnection = newConnectionProfile;
 			this.refreshConnections(newConnectionProfile);
-			await this._activeClientSession.updateConnection(this._activeConnection);
+			this._activeClientSession.updateConnection(this._activeConnection.toIConnectionProfile()).catch(() => {
+				let kernelDisplayName = this.getDisplayNameFromSpecName(this.clientSession.kernel.name);
```

Comment:
When this error happens, what is the result in terms of user experience? Do they get "select connection" shown?

Reply:
We need to cleanly handle this so that users know it failed and can try again somehow.

Reply:
Yes let's discuss further. I'm concerned about what happens next - will the Attach To dropdown make it clear that a) you're not connected, and b) you should be able to choose a new connection.